src/wrapper.test.js

```diff
@@ -179,220 +189,328 @@ describe('Distro initialization', () => {
         process.env.LUMIGO_DEBUG_SPANDUMP = '/dev/stdout';
       });
 
-      test('should initialize the FileSpanExporter', () => {
-        jest.isolateModules(async () => {
+      test('should initialize the FileSpanExporter', async () => {
+        await jest.isolateModulesAsync(async () => {
           const wrapper = jest.requireActual('./wrapper');
-          await wrapper.init.then(() => {
-            expect(FileSpanExporter).toHaveBeenCalledWith('/dev/stdout');
-          });
+          await wrapper.init;
+
+          expect(FileSpanExporter).toHaveBeenCalledWith('/dev/stdout');
         });
       });
     });
   });
 
   describe('outside of a computing platform for which we have detectors', () => {
+    beforeEach(() => {
+      process.env.LUMIGO_TRACER_TOKEN = LUMIGO_TRACER_TOKEN;
+      process.env.OTEL_SERVICE_NAME = 'service-1';
+      process.env.LUMIGO_REPORT_DEPENDENCIES = 'false';
+    });
+
     test('NodeTracerProvider should be given a resource with all the right attributes', async () => {
-      jest.isolateModules(async () => {
-        process.env.LUMIGO_TRACER_TOKEN = LUMIGO_TRACER_TOKEN;
-        process.env.OTEL_SERVICE_NAME = 'service-1';
+      await jest.isolateModulesAsync(async () => {
         const wrapper = jest.requireActual('./wrapper');
-        await wrapper.init
-          .then((initStatus) => initStatus.tracerProvider.resource)
-          .then((resource) => {
-            expect(resource.attributes['framework']).toBe('express');
-            expect(resource.attributes['service.name']).toBe('service-1');
-
-            checkBasicResourceAttributes(resource);
-          });
+        const { tracerProvider } = await wrapper.init;
+        const resource = tracerProvider.resource;
+
+        expect(resource.attributes['framework']).toBe('node');
+        expect(resource.attributes['service.name']).toBe('service-1');
+
+        checkBasicResourceAttributes(resource);
       });
     });
   });
 
   describe('On Amazon ECS', () => {
     beforeEach(() => {
+      process.env = { ...ORIGINAL_PROCESS_ENV };
       process.env.ECS_CONTAINER_METADATA_URI = ECS_CONTAINER_METADATA_URI;
+      process.env.LUMIGO_REPORT_DEPENDENCIES = 'false';
     });
 
     describe('without the Task Metadata V4 endpoint', () => {
-      test('NodeTracerProvider should be given a resource with all the right attributes', async () => {
-        jest.isolateModules(async () => {
-          process.env.LUMIGO_TRACER_TOKEN = LUMIGO_TRACER_TOKEN;
-          process.env.OTEL_SERVICE_NAME = 'service-1';
+      beforeEach(() => {
+        process.env.LUMIGO_TRACER_TOKEN = LUMIGO_TRACER_TOKEN;
+        process.env.OTEL_SERVICE_NAME = 'service-1';
+      });
 
+      test('NodeTracerProvider should be given a resource with all the right attributes', async () => {
```

Comment:
test with the same name - 2

Reply:
The hierarchy of the various levels of `describe` is different, no?